packages/docusaurus-theme-classic/src/theme/DocSidebarItem/Category/index.tsx

```diff
@@ -46,6 +46,22 @@ function useAutoExpandActiveCategory({
   }, [isActive, wasActive, collapsed, updateCollapsed]);
 }
 
+function onlyInactiveUnlistedLinks(
```

Comment:
What about having just `isInactiveUnlistedSidebarItem(item,activePath)`, generic and taking doc/category...?

I don't think there's value in encapsulating the `items.every()` here

move technical code to theme-common